lib/mempool/mempool.js

```diff
@@ -953,6 +967,72 @@ class Mempool extends EventEmitter {
         0);
     }
 
+    // If we reject RBF transactions altogether we can skip these checks,
+    // because incoming conflicts are already rejected as double spends.
+    let conflicts = [];
+    if (this.options.replaceByFee)
+      conflicts = this.getConflicts(tx);
+
+    if (conflicts.length) {
+      let conflictingFees = 0;
+      let totalEvictions = 0;
+
+      // Replacement TXs must pay a higher fee rate than each conflicting TX.
+      // We compare using deltaFee to account for prioritiseTransaction()
+      for (const conflict of conflicts) {
+        conflictingFees += conflict.descFee;
+        totalEvictions += this.countDescendants(conflict) + 1;
+
+        // Replacement TXs must not evict/replace more than 100 descendants
+        if (totalEvictions > 100) {
+          throw new VerifyError(tx,
+          'nonstandard',
+          'too many potential replacements',
+          0);
+        }
+
+        if (entry.getDeltaRate() <= conflict.getDeltaRate()) {
+          throw new VerifyError(tx,
+          'insufficientfee',
+          'insufficient fee',
+          0);
+        }
+      }
+
+      // Replacement TX must also pay for the total fees of all descendant
+      // transactions that will be evicted if an ancestor is replaced.
+      // Thus the replacement "pays for the bandwidth" of all the conflicts.
+      if (entry.deltaFee <= conflictingFees) {
```

Comment:
This check is not strictly necessary as the next check covers this as well. If deltaFee <= conflictingFees the `feeRemaining` will be negative.